File: typescript/scrabble-score/scrabble-score.ts
const letterValues = new Map<string, number>(Object.entries({
  "AEIOULNRST": 1,
  "DG": 2,
  "BCMP": 3,
  "FHVWY": 4,
  "K": 5,
  "JX": 8,
  "QZ": 10,
}).flatMap(([letters, score]) => [...letters].map((letter) => [letter, score])));

export function score(word: string | undefined): number {
  return [...(word || '').toUpperCase()]
    .map((letter) => letterValues.get(letter)!)
    .reduce((a, b) => a + b, 0);
}

File: clojure/scrabble-score/src/scrabble_score.clj
(ns scrabble-score
  (:require [clojure.string :as string]))

(def ^:private values-for-letters
  {1 [\A \E \I \O \U \L \N \R \S \T]
   2 [\D \G]
   3 [\B \C \M \P]
   4 [\F \H \V \W \Y]
   5 [\K]
   8 [\J \X]
   10 [\Q \Z]})

(def ^:private letter-values
  (into {} (for [[score letters] values-for-letters
                 letter letters]
             [letter score])))

(defn score-word [word]
  (->> word
       string/upper-case
       (map letter-values)
       (reduce +)))

(defn score-letter [letter]
  (score-word letter))
